internal/api/impl/proxy/proxy.go

```diff
@@ -379,3 +387,250 @@ func (h *httpProxy) prepareTLSConfig() (*tls.Config, error) {
 	}
 	return secretModel.BuildTLSConfig(h.secret.TLSConfig)
 }
+
+// proxy SQL queries provided the plugin sends the SQL query in the request body as
+// {"query": "select * from table"}
+func newSQLProxy(spec v1.DatasourceSpec, path string, crypto crypto.Crypto, retrieveSecret func(name string) (*v1.SecretSpec, error)) (proxy, error) {
+	cfg := &datasourceSQL.Config{}
+	err := datasourceproxy.ValidateAndExtract(datasourceSQL.SQLProxyKindName, spec.Plugin.Spec, cfg)
+	if err != nil {
+		logrus.WithError(err).Error("unable to build or find the sql config in the datasource")
+		return nil, echo.NewHTTPError(http.StatusBadGateway, "unable to find the sql config")
+	}
+
+	var scrt *v1.SecretSpec
+	if len(cfg.Secret) > 0 {
+		scrt, err = retrieveSecret(cfg.Secret)
+		if err != nil {
+			return nil, err
+		}
+		if decryptErr := crypto.Decrypt(scrt); decryptErr != nil {
+			logrus.WithError(decryptErr).Errorf("unable to decrypt the secret")
+			return nil, apiinterface.InternalError
+		}
+	}
+
+	return &sqlProxy{
+		config: cfg,
+		secret: scrt,
+		path:   path,
+	}, nil
+}
+
+type sqlQuery struct {
+	Query string `json:"query"`
+}
+
+type sqlProxy struct {
+	config   *datasourceSQL.Config
+	secret   *v1.SecretSpec
+	path     string
+	password string
+}
+
+func (s *sqlProxy) serve(c echo.Context) error {
+	r := c.Request()
+
+	if s.path != "query" {
+		return apiinterface.HandleBadRequestError("SQL proxy only supports the '/query' path")
+	}
+
+	// if this isn't a POST request don't perform the SQL query
+	if r.Method != http.MethodPost {
+		return echo.NewHTTPError(http.StatusMethodNotAllowed, fmt.Sprintf("you are not allowed to use this endpoint %q with the HTTP method %s", s.path, r.Method))
+	}
+
+	q := &sqlQuery{}
```

Comment:
probably at some point we will need to read/parse/analyse the query to ensure Perses cannot be used in order to write data in the targeted database.

That's not something we should do in this PR, but something to think for the next one